File: javascript/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.js
/**
 * @param {number} x
 * @param {number} y
 * @param {number[][]} points
 * @return {number}
 */
 var nearestValidPoint = function(x, y, points) {
    let min = Number.MAX_SAFE_INTEGER
    let idx = -1
    for (let i = 0; i < points.length; i++) {
        if (x === points[i][0]) {
            const d = Math.abs(y - points[i][1])
            if (d < min) {
                min = d
                idx = i
            }
            continue
        }
        if (y === points[i][1]) {
            const d = Math.abs(x - points[i][0])
            if (d < min) {
                min = d
                idx = i
            }
        }
    }
    return idx
};

File: golang/1779-find-nearest-point-that-has-the-same-x-or-y-coordinate.go
package golang

import "math"

func nearestValidPoint(x int, y int, points [][]int) int {
    idx, min, n := -1, math.MaxInt64, len(points)
    for i := 0; i < n; i++ {
        if x == points[i][0] {
            d := int(math.Abs(float64(y - points[i][1])))
            if d < min {
                min = d
                idx = i
            }
            continue
        }
        if y == points[i][1] {
            d := int(math.Abs(float64(x - points[i][0])))
            if d < min {
                min = d
                idx = i
            }
        }
    }
    return idx
}